code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,305 @@
+
+/datum/action/innate/elite_attack
+	name = "Elite Attack"
+	icon_icon = 'icons/mob/actions/actions_elite.dmi'
+	button_icon_state = ""
+	var/mob/living/simple_animal/hostile/asteroid/elite/M
+	var/chosen_message
+	var/chosen_attack_num = 0
+
+/datum/action/innate/elite_attack/Grant(mob/living/L)
+	if(istype(L, /mob/living/simple_animal/hostile/asteroid/elite))
+		M = L
+		return ..()
+	return FALSE
+
+/datum/action/innate/elite_attack/Activate()
+	M.chosen_attack = chosen_attack_num
+	to_chat(M, chosen_message)
+	
+/mob/living/simple_animal/hostile/asteroid/elite/Life()
+	. = ..()
+	if(isturf(loc))
+		for(var/obj/structure/elite_crevice/crevice in loc)
+			if(crevice == myparent && myparent.activity == 2)
+				adjustHealth(-maxHealth*0.05)
+				var/obj/effect/temp_visual/heal/H = new /obj/effect/temp_visual/heal(get_turf(src))
+				H.color = "#FF0000"
+	if(myparent)
+		if(myparent.activity == 1 && myparent.activator.stat == DEAD)
+			myparent.onEliteWon()
+
+/mob/living/simple_animal/hostile/asteroid/elite/death()
+	. = ..()
+	if(myparent)
+		myparent.onEliteLoss()
+
+//The Glowing Crevice, the actual "spawn-point" of elites, handles the spawning, arena, and procs for dealing with basic scenarios.
+
+/obj/structure/elite_crevice
+	name = "glowing crevice"
+	desc = "A glowing, red hole which doesn't seem to have a bottom.  You feel pressured to reach your hand out towards it..."
+	armor = list("melee" = 100, "bullet" = 100, "laser" = 100, "energy" = 100, "bomb" = 100, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100)
+	max_integrity = 10000
+	var/activity = 0
+	var/boosted = 0
+	var/times_won = 0
+	var/mob/living/carbon/human/activator = null
+	var/mob/living/simple_animal/hostile/asteroid/elite/mychild = null
+	var/potentialspawns = list(/mob/living/simple_animal/hostile/asteroid/elite/goliath)
+	icon = 'icons/mob/lavaland/elite_lavaland_monsters.dmi'
+	icon_state = "elite_crevice"
+	light_color = LIGHT_COLOR_RED
+	light_range = 3
+	anchored = TRUE
+	density = FALSE
+	
+/obj/structure/elite_crevice/attack_hand(mob/user)
+	if(activity == 0)
+		if(istype(user, /mob/living/carbon/human))
+			activity = 1
+			var/mob/dead/observer/elitemind = null
+			visible_message("<span class='boldwarning'>The crevice glows.  Your instincts tell you to step back.</span>")
+			activator = user
+			if(boosted == 1)
+				visible_message("<span class='boldwarning'>Something within the crevice stirs...</span>")
+				var/list/candidates = pollCandidatesForMob("Do you want to play as a lavaland elite?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, src, POLL_IGNORE_SENTIENCE_POTION)
+				if(candidates.len)
+					visible_message("<span class='boldwarning'>The stirring sounds increase in volume!</span>")
+					elitemind = pick(candidates)
+					SEND_SOUND(elitemind, sound('sound/effects/magic.ogg'))
+					to_chat(elitemind, "<b>You have been chosen to play as a Lavaland Elite.</b>")
```

Comment:
make this one to_chat. to_chat spam lags the client

Reply:
Done for upcoming fix.